### ISSUE TITLE
The option --FORMAT_GEO_MAP is ignored for some combinations of -JX... and -f...

### ISSUE THREAD

**Issue by uleysky** [GenericMappingTools/gmt#2434]
Command one:
`
gmt psbasemap -R0/1/40/50 -JX10c/10cd -BSWne -Bx0.5 -By2 -P -f0f,1y,2f --FORMAT_GEO_MAP=DG >tstlat.ps
`
Result: latitude is set on the y-axis with correct suffix.

Command two:
`
gmt psbasemap -R0/1/40/50 -JX10c/10cd -BSWne -Bx0.5 -By2 -P -f0f,1x,2f --FORMAT_GEO_MAP=DG >tstlon.ps
`
Result: longitude is set on the y-axis (degree sign is present), but --FORMAT_GEO_MAP is ignored (no suffix).

GMT version: 6.1.0_e53ace7-dirty_2020.01.10 (e53ace7)

**Comment by PaulWessel** [comment]
Acknowledging this is indeed what happens.  Not sure if we have a solution for this yet.

**Comment by seisman** [comment]
**-f** specifies the data types of input/output. It's a little weird to see that **basemap** can accept **-f** option.

**Comment by PaulWessel** [comment]
Yes, but if the above instead was **psxy** then it would not be so weird.  While the **-J** could tell us that one coordinate is geographic and the other is Cartesian, it _cannot_ tell us if it is longitude or latitude that is given, which is where **-f** comes in handy.  I also need to make plots like this from time to time (something as a function of lon or lat, or the inverse) so I would like to come up with a solution.

**Comment by PaulWessel** [comment]
Perhaps Dr. Regex can tell me how to turn

`" GMT_def(X)"`

into

`"* GMT_def_scale(X), GMT_def_unit(X)"`
?

**Comment by seisman** [comment]
Try the command below. It assumes that _X_ is letters and underscores only.
```
sed -i "" -E 's/ GMT_def\(([a-zA-Z_]+)\)/ * GMT_def_scale\(\1\), GMT_def_unit\(\1\)/' gmt_init.c
```

**Comment by PaulWessel** [comment]
Thanks, seems to have worked.  Just running tests before making a commit.
